### PR TITLE
Reupload Assignment 4

### PR DESCRIPTION
**Pull request by AlexRouge**
Unfortunately, speaking with some of my colleagues, I only realized today that my previous pull request dated October 9th was not only empty but closed by itself even though I was sure I had done things correctly. I understand very well if assignment 4 is not accepted now but I assure you that I've done everything before the deadline. I hope you can understand, have a good evening

### CODE REVIEW

**Comment by dgarijo** [comment]
No worries. However you are uploading notebooks, which does not follow the
instructions shown in class. Please upload the correct files ( see the
slides)

**Comment by AlexRouge** [comment]
Now they should have been uploaded in the correct format; thank you again for your comprehension

File: Assignment4/.DS_Store
Comment (dgarijo) [inline Assignment4/.DS_Store:1]: Remove this file

File: Assignment4/AlexRouge - CA43899MT/task07.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+q1 = prepareQuery('''
+  SELECT ?Subclass WHERE {
+    ?Subclass rdfs:subClassOf/rdfs:subClassOf* ns:LivingThing.
+  }
+  ''',
+  initNs = { "rdfs":RDFS, "ns":ns}
+)
+# Visualize the results
+print("RDFLib")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/AlexRouge - CA43899MT/task07.py:39]: This does not provide the same results as the previous sparql function. Use a recursive function

File: Assignment4/AlexRouge - CA43899MT/task07.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  SELECT ?person WHERE {
+    ?subclass rdfs:subClassOf* ns:Person.
+    ?person rdf:type ?subclass.
+  }
+  ''',
+  initNs = { "rdf":RDF, "ns":ns, "rdfs":RDFS}
+)
+# Visualize the results
+print("RDFLib")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
```
Comment (dgarijo) [inline Assignment4/AlexRouge - CA43899MT/task07.py:59]: Same problem as above

File: Assignment4/AlexRouge - CA43899MT/task07.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      FILTER(?other1 != ?other2) .
+    }
+  ''',
+  initNs={"rdf": RDF, "ns": ns, "rdfs": RDFS, "foaf": foaf}
+)
+
+# Execute the query and print the results
+print("Persons who know at least two entities:")
+for row in g.query(q2):
+  print(row)
```
Comment (dgarijo) [inline Assignment4/AlexRouge - CA43899MT/task07.py:144]: Missing 7.4 and 7.5 in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address remaining issues and remove the ds store file

**Review by dgarijo — APPROVED** [review]

**Pull request merged**